File: 021 Merge Two Sorted Lists/merge_two_sorted_lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {	// Author: Xinyu Chen
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode* current = new ListNode(0);
        ListNode* prehead = current;
        while (l1 != nullptr && l2 != nullptr) {
            if (l1->val < l2->val) {
                current->next = l1;
                l1 = l1->next;
            } else {
                current->next = l2;
                l2 = l2->next;
            }
            current = current->next;
        }
        if (l1 != nullptr) current->next = l1;
        else if (l2 != nullptr) current->next = l2;
        return prehead->next;
    }
};

File: 021 Merge Two Sorted Lists/merge_two_sorted_lists.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution { // Author: Xinyu Chen
    public ListNode MergeTwoLists(ListNode l1, ListNode l2)
    {
        ListNode result = new ListNode(0);
        ListNode curr = result;
        while (l1 != null && l2 != null)
        {
            if (l1.val < l2.val)
            {
                curr.next = l1;
                l1 = l1.next;
            }
            else
            {
                curr.next = l2;
                l2 = l2.next;
            }
            curr = curr.next;
        }
        if (l1 != null) curr.next = l1;
        else curr.next = l2;
        return result.next;
    }
}
